.github/workflows/auto-update-contributors-readme.yml

```diff
@@ -0,0 +1,99 @@
+        id: default_branch
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          defaultBranch="$( \
+            gh api /repos/${{ github.repository }} \
+              --method GET \
+              --jq '.default_branch'
+          )"
+
+          isProtected="$( \
+            gh api /repos/${{ github.repository }}/branches/$defaultBranch \
+            --jq '.protected'
+          )"
+
+          echo "::set-output name=is_protected::$isProtected"
+
+      - name: Check if a pull request already exists
+        if: ${{ steps.default_branch.outputs.is_protected == 'true' }}
+        id: find_existing_pr
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ID_PR="$( \
+            gh api search/issues \
+              --method GET \
+              --raw-field q='repo:${{ github.repository }} type:pr state:open "Auto-update list of contributors" in:title' \
+              --jq '.items.[0].number' \
+          )"
+
+          echo "::set-output name=id_pr_to_close::$ID_PR"
+
+      - name: Close the existing pull request
+        if: ${{ steps.find_existing_pr.outputs.id_pr_to_close }}
+        id: close_existing_pr
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh pr close ${{ steps.find_existing_pr.outputs.id_pr_to_close }} \
+            --repo ${{ github.repository }} \
+            --delete-branch
+
+      # https://github.com/akhilmhdh/contributors-readme-action
+      - name: Update the list of contributors in the README file
+        uses: akhilmhdh/contributors-readme-action@v2.3
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          commit_message: 'chore: auto-update contributors list (GH Action)'
+          committer_username: 'akhilmhdh/contributors-readme-action'
+          is_protected: ${{ steps.default_branch.outputs.is_protected == 'true' }}
+          use_username: true
+
+      # The Action `akhilmhdh/contributors-readme-action` creates PRs with a fixed title:
+      # https://github.com/akhilmhdh/contributors-readme-action/blob/v2.3/src/index.js#L169
+      - name: Rename the new PR with a better title
+        if: ${{ steps.default_branch.outputs.is_protected == 'true' }}
+        id: rename_new_pr
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          sleep 10 # Some delay is needed to make sure GitHub had time to create the new PR
+
+          ID_PR="$( \
+            gh api search/issues \
+              --method GET \
+              --raw-field q='repo:${{ github.repository }} type:pr state:open "contributors readme action update" in:title' \
+              --jq '.items.[0].number' \
+          )"
+
+          gh pr edit $ID_PR \
+            --repo ${{ github.repository }} \
+            --title "Auto-update list of contributors"
+
+          echo "::set-output name=id_new_pr::$ID_PR"
```

Comment:
This whole step may not be needed in the future:
https://github.com/akhilmhdh/contributors-readme-action/pull/35

Instead we will be able to do pass an input to the step responsible for creating the PR:

```yaml
      - uses: akhilmhdh/contributors-readme-action@v2.3
        with:
          pr_title: My Custom Title
```

Plus, the GitHub Action `akhilmhdh/contributors-readme-action` may be able to output the ID of the PR that was created.

Reply:
All removed!